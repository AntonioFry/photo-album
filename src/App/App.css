* { 
  box-sizing: border-box;
}

main {
  background-color: snow;
  min-height: 668px;
}

header {
  background-color: black;
  padding: 20px;
}

.title-heading {
  margin: 0px;
  color: white;
}

