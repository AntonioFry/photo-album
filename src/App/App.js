import React, { Component } from 'react';
import './App.css';
import ImageInspector from '../ImageInspector/ImageInspector';
import Gallery from '../Gallery/Gallery';

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentPhotoAlbum: [],
    }
  }

  setCurrentPhotoAlbum = (photos) => {
    this.setState({ currentPhotoAlbum: photos })
  };

  render() {
    return (
      <main>
        <header>
          <h1 data-testid="title" className="title-heading">My Photo Album</h1>
        </header>
        <Gallery setCurrentPhotoAlbum={this.setCurrentPhotoAlbum} />
        <ImageInspector imageCollection={this.state.currentPhotoAlbum} />
      </main>
    );
  }
}

export default App;