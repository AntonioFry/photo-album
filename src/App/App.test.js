import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';

describe('App', () => {

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('title')).toHaveTextContent('My Photo Album');
  });

})
