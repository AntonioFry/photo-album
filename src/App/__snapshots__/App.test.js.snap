// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
<main>
  <header>
    <h1
      className="title-heading"
      data-testid="title"
    >
      My Photo Album
    </h1>
  </header>
  <section
    className="gallery-section"
  >
    <article
      className="photo-card"
      style={
        Object {
          "backgroundImage": "url(ants-and-plants.JPG)",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="photo-comment-container"
      >
        <p
          className="photo-comment"
          data-testid="photo-comment-1"
        >
          This is one the first image I took of any sort of insect. After being able to see how much detail that was visible in the image versus the naked eye, I decided to try taking more of the small critters that roam the earth.
        </p>
        <div
          className="photo-card-btn-container"
        >
          <button
            className="album-name-btn"
            onClick={[Function]}
          >
            Insect Photos
          </button>
        </div>
      </div>
    </article>
    <article
      className="photo-card"
      style={
        Object {
          "backgroundImage": "url(jewell.JPG)",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="photo-comment-container"
      >
        <p
          className="photo-comment"
          data-testid="photo-comment-2"
        >
          This is nice shot of a downhill street in my neighborhood that faces the mountains. However I would say it looks a lot nicer when actually there.
        </p>
        <div
          className="photo-card-btn-container"
        >
          <button
            className="album-name-btn"
            onClick={[Function]}
          >
            Environment Photos
          </button>
        </div>
      </div>
    </article>
    <article
      className="photo-card"
      style={
        Object {
          "backgroundImage": "url(deer-shedding-velvet.JPG)",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="photo-comment-container"
      >
        <p
          className="photo-comment"
          data-testid="photo-comment-3"
        >
          Took this shot at the Rocky Mountain Arsenal. Though there were many deer, this was the only one I saw who was shedding the velvet from their antlers
        </p>
        <div
          className="photo-card-btn-container"
        >
          <button
            className="album-name-btn"
            onClick={[Function]}
          >
            Wildlife Photos
          </button>
        </div>
      </div>
    </article>
    <article
      className="photo-card"
      style={
        Object {
          "backgroundImage": "url(grey-jay.JPG)",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="photo-comment-container"
      >
        <p
          className="photo-comment"
          data-testid="photo-comment-4"
        >
          Took this shot up at Echo Lake when the bird landed meer feet away from me. It wasn't until I searched up th bird I realized it was a grey jay.
        </p>
        <div
          className="photo-card-btn-container"
        >
          <button
            className="album-name-btn"
            onClick={[Function]}
          >
            Bird Photos
          </button>
        </div>
      </div>
    </article>
  </section>
</main>
`;
